alacritty_terminal/src/grid/mod.rs

```diff
@@ -72,6 +72,37 @@ pub trait GridCell {
 }
 
 /// Represents the terminal display contents
+///
+/// +--------------------------------------------+ <--- ???
+/// |                                            |
+/// |                                            |
+/// |                                            |
+/// |                                            |
+/// |                                            |
+/// |              SCROLLBACK REGION             |
+/// |                                            |
+/// |                                            |
+/// |                                            |
+/// |                                            |
+/// |                                            |
+/// +--------------------------------------------+ <--- ???
+/// |                                            |
+/// |              ABOVE SCROLLING               |
+/// |                   REGION                   |
+/// |                                            |
+/// +--------------------------------------------+ <--- ???
+/// |                                            |
+/// |              SCROLLING REGION              |
+/// |                                            |
+/// +--------------------------------------------+ <--- display_offset
+/// |                                            |
+/// |              BELOW SCROLLING               |
+/// |                   REGION                   |
+/// |                                            |
+/// +--------------------------------------------+ <--- raw.zero
+///                                              |
+///                                              v
+///                                             cols
```

Comment:
I think if we want to add this documentation comment, we need to add a few more words to explain exactly what's going on and remove the `???` parts.

Additionally we should make sure it formats properly, which will likely require putting this in a codeblock.